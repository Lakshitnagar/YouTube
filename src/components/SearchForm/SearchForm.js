import React, {useState} from 'react';
import search from '../../assets/images/search_18dp.svg';
import './SearchForm.scss';
import {fetchYouTubeVideosByKeyword} from "../../store/actions/YouTubeActions";
import PropTypes from "prop-types";
import {connect} from "react-redux";

export function SearchForm(props) {
    const [searchKeyword, setSearchKeyword] = useState('');

    const onSearchFormSubmit = (e) => {
        e.preventDefault();
        if (searchKeyword) props.fetchYouTubeVideosByKeyword(searchKeyword);
    };

    return (
        <div className="searchForm">
            <form className="searchForm__form" onSubmit={(e) => {
                onSearchFormSubmit(e)
            }}>
                <div className="searchForm__inputContainer">
                    <input id="search" autoCapitalize="none" autoComplete="off" autoCorrect="off"
                           type="text" spellCheck="false" placeholder="Search"
                           onChange={e => setSearchKeyword(e.target.value)}/>
                </div>

                <button id="search-icon" className="searchForm__submit" aria-label="Search">
                    <img src={search} alt='search'/>
                </button>
            </form>
        </div>
    );
}

export const mapDispatchToProps = (dispatch) => {
    return {
        fetchYouTubeVideosByKeyword: (keyword) => dispatch(fetchYouTubeVideosByKeyword(keyword))
    };
};

SearchForm.propTypes = {
    fetchYouTubeVideosByKeyword: PropTypes.func,
};

export default connect(null, mapDispatchToProps)(SearchForm);
