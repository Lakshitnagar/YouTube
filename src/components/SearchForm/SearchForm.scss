.searchForm {
  width: 33%;

  .searchForm__form {
    display: flex;

    .searchForm__inputContainer {
      border: solid 1px #ccc;
      border-right: none;
      border-radius: 2px 0 0 2px;
      padding: 2px 6px;
      width: 100%;

      input {
        width: 100%;
        border: none;
        outline: none;
        background-color: transparent;
        box-shadow: none;
        color: inherit;
        font-family: 'Roboto', 'Noto', sans-serif;
        font-size: 16px;
        font-weight: 400;
        line-height: 24px;
        margin-left: 4px;
        max-width: 100%;
        text-align: left;
      }
    }

    .searchForm__submit {
      cursor: pointer;
      width: 65px;
      border: 1px solid #d3d3d3;
      background-color: #f8f8f8;
      border-radius: 0 2px 2px 0;
      margin: 0;
    }
  }
}
