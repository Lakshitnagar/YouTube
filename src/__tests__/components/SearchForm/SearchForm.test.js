import React from 'react';
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import SearchForm from "../../../components/SearchForm/SearchForm";

jest.mock('../../../assets/images/search_18dp.svg', () => 'mockSearchIcon');

describe('SearchForm', function () {
    it('should render and match snapshot', function () {
        const component = <SearchForm />;

        const wrapper = shallow(component);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
