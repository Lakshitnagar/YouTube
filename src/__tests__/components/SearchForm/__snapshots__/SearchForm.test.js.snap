// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render and match snapshot 1`] = `
<div
  className="searchForm"
>
  <form
    className="searchForm__form"
  >
    <div
      className="searchForm__inputContainer"
    >
      <input
        autoCapitalize="none"
        autoComplete="off"
        autoCorrect="off"
        id="search"
        name="search_query"
        placeholder="Search"
        spellCheck="false"
        tabIndex="0"
        type="text"
      />
    </div>
    <button
      aria-label="Search"
      className="searchForm__submit"
      id="search-icon"
    >
      <img
        alt="search"
        src="mockSearchIcon"
      />
    </button>
  </form>
</div>
`;
